### PR TITLE
Bump up to go1.22

### PR DESCRIPTION
**Pull request by reasonerjt**
Bump up the golang version to v1.22 

It also updates controller-gen to v0.14.0 to fix an issue under new
version of go.
More details see https://github.com/golang/go/issues/65637

It's known that it causes decrease to the coverage on codecov due to the change in go1.22 to calculate coverage

# Please indicate you've done the following:

- [X] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: hack/build-image/Dockerfile [hunk 1]
```diff
@@ -30,7 +30,7 @@ RUN wget --quiet https://github.com/kubernetes-sigs/kubebuilder/releases/downloa
     chmod +x /usr/local/kubebuilder/bin/kubebuilder
 
 # get controller-tools
-RUN go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.12.0
+RUN go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.14.0
```
Comment (mateusoliveira43) [inline hack/build-image/Dockerfile:33]: why this upgrade of controller-gen?

File: hack/test.sh [hunk 2]
```diff
@@ -42,5 +43,5 @@ fi
 # but the user and group don't exist inside the container, when the code(https://github.com/kubernetes-sigs/controller-runtime/blob/v0.10.2/pkg/internal/testing/addr/manager.go#L44)
 # tries to get the cache directory, it gets the directory "/" and then get the permission error when trying to create directory under "/".
 # Specifying the cache directory by environment variable "XDG_CACHE_HOME" to workaround it
-XDG_CACHE_HOME=/tmp/ go test -installsuffix "static" -short -timeout 1200s -coverprofile=coverage.out "${TARGETS[@]}"
+#XDG_CACHE_HOME=/tmp/ go test -installsuffix "static" -short -timeout 1200s -coverprofile=coverage.out "${TARGETS[@]}"
```
Comment (mateusoliveira43) [inline hack/test.sh:46]: is not this change DANGEROUS? are not test DISABLED now?

File: hack/test.sh [hunk 3]
```diff
@@ -21,7 +21,8 @@ set -o pipefail
 
 export CGO_ENABLED=0
 
-TARGETS=($(go list ./pkg/... | grep -v "github.com/vmware-tanzu/velero/pkg/builder"))
+TARGETS=($(go list ./pkg/... | grep -v "github.com/vmware-tanzu/velero/pkg/builder" | grep -v "mocks" |
```
Comment (mateusoliveira43) [inline hack/test.sh:24]: I would simplify command and be more explicit about what to exclude

all output lines of command will containe `github.com/vmware-tanzu/velero/pkg/`, so would remove long string

```sh
go list ./pkg/... | grep -vE 'pkg/builder|pkg/apis|pkg/test|pkg/generated|pkg/plugin/generated|mocks'
```

**Review by mateusoliveira43 — CHANGES_REQUESTED** [review]
left comments

File: hack/test.sh (hunk 3 above)
Comment (mateusoliveira43) [reply hack/test.sh:24]: also, add task here https://github.com/vmware-tanzu/velero/issues/6190 to remove excludes not needed after code removal

**Comment by blackpiglet** [comment]
The UT is impacted. Please double-check the change mentioned by @mateusoliveira43.

File: hack/build-image/Dockerfile (hunk 1 above)
Comment (reasonerjt) [reply hack/build-image/Dockerfile:33]: Please see the commit message:
```
It also updates controller-gen to v0.14.0 to fix an issue under new
versino of go.
More details see https://github.com/golang/go/issues/65637
```

**Comment by reasonerjt** [comment]
@blackpiglet 
> The UT is impacted. Please double-check the change mentioned by @mateusoliveira43.

The coverage may have to be impacted, b/c the go1.22 stopped skipping the packages that do not include  `_test` files, I find it also changed in other details, for example, it stopped taking into some inline comments as covered.

File: hack/test.sh (hunk 2 above)
Comment (reasonerjt) [reply hack/test.sh:46]: This PR was changed to "ready for review" by mistake, I was still investigating the change in coverage caused by go1.22.

File: hack/test.sh (hunk 3 above)
Comment (reasonerjt) [reply hack/test.sh]: Sure. thanks

File: hack/build-image/Dockerfile (hunk 1 above)
Comment (mateusoliveira43) [reply hack/build-image/Dockerfile:33]: thank for the explanation!

would add that information to the PR description, it is more visible to reviewers

**Review by blackpiglet — APPROVED** [review]

**Review by ywk253100 — APPROVED** [review]

**Pull request merged**